### DISCUSSION TITLE
SRX Talon for Beta 2024

### DISCUSSION THREAD

**Discussion by kglasnapp** [wpilibsuite/2024Beta#42] [Q&A] [answered]
Has anyone gotten SRX Talons working with the 2024 beta software? I have converted our code to the 2024 beta and when I try and run the code -- we get a log stating we new FW Version 6. I can't seem to find 2024 firmware for the SRX Talon. I can find it for the Talon FX.

**Comment by CoryNessCTR** [comment]
Talon SRXs use Phoenix 5 firmware (22.x) and Phoenix 5 API.
So make sure you include the Phoenix 5 vendordep and use Phoenix 5 API in your robot program.

It sounds like you're using the TalonFX class from Phoenix 6 to try to control Talon SRXs, which would explain your symptoms.

**Reply by kglasnapp** [reply, marked as answer]
Thanks so much -- after using Phoenix 5 it works. I still needed to include Phoenix 6 in vendordeps?

**Reply by JCaporuscio** [reply]
>  I still needed to include Phoenix 6 in vendordeps?

Yep, that's expected.  Phoenix 5 and 6 share a common set of low-level tools/libraries that should only be included once in your project.  Rather than have the triple vendordep (5, 6, or 5-and-6) setup from last year or have a third, separate driver vendorlib we chose to bundle the dependencies with Phoenix 6.

That keeps the number of vendordeps you need to a minimum and lets folks who are only using newer products use just the Phoenix 6 vendordep.
